src/test/java/net/oauth/jsontoken/JsonTokenParserTest.java

```diff
@@ -227,4 +197,45 @@ public void testPublicKey() throws Exception {
       // expected
     }
   }
+
+  private boolean verifyTimeFrame(Instant issuedAt, Instant expiration) throws Exception {
+    HmacSHA256Signer signer = new HmacSHA256Signer("google.com", "key2", SYMMETRIC_KEY);
+    JsonToken token = new JsonToken(signer, clock);
+    if (issuedAt != null) {
+      token.setIssuedAt(issuedAt);
+    }
+    if (expiration != null) {
+      token.setExpiration(expiration);
+    }
+    token.setAudience("http://www.google.com");
+
+    JsonTokenParser parser = new JsonTokenParser(clock, locators, new IgnoreAudience());
+    JsonToken checkToken = parser.deserialize(token.serializeAndSign());
+    try {
+      parser.verify(checkToken);
+      return true;
+    } catch (IllegalStateException e) {
```

Comment:
For failure tests can you use assertThrows?

Reply:
Sorry this comment should be on the next two methods instead.

Reply:
`assertThrows` requires JUnit5, and replacing JUnit4 would require an additional refactor for other unrelated tests since JUnit5 no longer allows for extending TestCase (which the existing classes use). I can update the test cases in another PR.

Reply:
Actually, JUnit4 should be sufficient. Either way, the upgrade and subsequent changes will be in a new PR so I can update other unrelated tests as well.